pkg/arvo/app/permission-hook.hoon

```diff
@@ -0,0 +1,282 @@
+::  permission-hook: allows mirroring permissions between local and foreign
+::  ships. access control to an owned permission path is specified by the
+::  access-control path.
+::
+/-  *permission-hook
+/+  *permission-json
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff [%permission-update permission-update]]
+      [%quit ~]
+      [%poke wire dock [%permission-action permission-action]]
+      [%pull wire dock ~]
+      [%peer wire dock path]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  owner-access  [ship=ship access-control=path]
+::
++$  state-zero
+  $:  synced=(map path owner-access)
+      access-control=(map path (set path))
+      boned=(map wire (list bone))
+  ==
+::
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  [~ ?~(old this this(+<+ u.old))]
+::
+++  poke-permission-hook-action
+  |=  act=permission-hook-action
+  ^-  (quip move _this)
+  ?-  -.act
+      %add-owned
+    ?>  (team:title our.bol src.bol)
+    ?:  (~(has by synced) owned.act)
+      [~ this]
+    =.  synced  (~(put by synced) owned.act [our.bol access.act])
+    =/  access-paths
+      ?.  (~(has by access-control) access.act)
+        [owned.act ~ ~]
+      (~(put in (~(got by access-control) access.act)) owned.act)
+    =.  access-control
+      (~(put by access-control) access.act access-paths)
+    =/  perm-path  [%permission owned.act]
+    :_  (track-bone perm-path)
+    [ost.bol %peer perm-path [our.bol %permission-store] perm-path]~
+  ::
+      %add-synced
+    ?>  (team:title our.bol src.bol)
+    =/  perm-path  [%permission path.act]
```

Comment:
I'd put this closer to the move construction down below.

Reply:
Will do, thanks.

Reply:
done.